adoc/admin-security-certificates.adoc

```diff
@@ -0,0 +1,208 @@
+
+Certificates are located in `/etc/kubernetes/pki` on each master nodes. Three root CA required for {productname} are
+
+* {kube} apiserver: path `/etc/kubernetes/pki/ca.{crt.key}`
+* front-end proxy: path `/etc/kubernetes/pki/front-proxy-ca.{crt.key}`
+* etcd: path `/etc/kubernetes/pki/etcd/ca.{crt.key}`
+
+The following certificates are managed by `kubeadm`.
+
+[%header,cols=4*]
+|===
+|Common Name
+|Parent CA
+|Path
+|Kind
+
+|kubernetes
+|
+|ca.crt,key
+|CA
+
+|kube-apiserver
+|kubernetes
+|apiserver.crt,key
+|Server
+
+|kube-apiserver-etcd-client
+|kubernetes
+|apiserver-etcd-client.crt,key
+|Client
+
+|kube-apiserver-kubelet-client
+|kubernetes
+|apiserver-kubelet-client.crt,key
+|Client
+
+|front-proxy-ca
+|
+|front-proxy-ca.crt,key
+|CA
+
+|front-proxy-client
+|front-proxy-ca
+|front-proxy-client.crt,key
+|Client
+
+|etcd-ca
+|
+|etcd/ca.crt,key
+|CA
+
+|kube-etcd-healthcheck-client
+|etcd-ca
+|etcd/healthcheck-client.crt,key
+|Client
+
+|kube-etcd-peer
+|etcd-ca
+|etcd/peer.crt,key
+|Server,Client
+
+|kube-etcd-server
+|etcd-ca
+|etcd/server.crt,key
+|Server,Client
+|===
+
+The following certificates are created by `skuba` and are stored in `Secret` resource.
+
+[%header,cols=4*]
+|===
+|Common Name
+|Parent CA
+|Secret Resource Name
+|Kind
+
+|oidc-dex
+|kubernetes
+|oidc-dex-cert
+|Server
+
+|oidc-gangway
+|kubernetes
+|oidc-gangway-cert
+|Server
+
+|cilium-etcd-client
+|etcd-ca
+|cilium-secret
+|Client
+|===
+
+== Deployment with custom root CA certificate
+
+Administrators can provide custom root CA certificates during cluster deployment and administrators can decide which root CA components to replace or replace all.
+
+After `skuba cluster init`, go to the generated folder. Create a `pki` folder and put your custom root CA certificate into `pki` folder.
+
+* Replace {kube} apiserver root CA certificate
++
+[source,bash]
+----
+mkdir -p my-cluster/pki
+cp <custom-apiserver-rootca-cert-path> my-cluster/pki/ca.crt
+cp <custom-apiserver-rootca-key-path> my-cluster/pki/ca.key
+chmod 644 my-cluster/pki/ca.crt
+chmod 600 my-cluster/pki/ca.key
+----
+
+* Replace front-end proxy root CA certificate
++
+[source,bash]
+----
+mkdir -p my-cluster/pki
+cp <custom-frontproxy-rootca-cert-path> my-cluster/pki/front-proxy-ca.crt
+cp <custom-frontproxy-rootca-key-path> my-cluster/pki/front-proxy-ca.key
+chmod 644 my-cluster/pki/front-proxy-ca.crt
+chmod 600 my-cluster/pki/front-proxy-ca.key
+----
+
+* Replace etcd root CA certificate
++
+[source,bash]
+----
+mkdir -p my-cluster/pki/etcd
+cp <custom-etcd-rootca-cert-path> my-cluster/pki/etcd/ca.crt
+cp <custom-etcd-rootca-key-path> my-cluster/pki/etcd/ca.key
+chmod 644 my-cluster/pki/etcd/ca.crt
+chmod 600 my-cluster/pki/etcd/ca.key
+----
+
+Then bootstrap the cluster `skuba node bootstrap`.
+
+== Automatic certificate renewal
+
+{productname} renews all the certificates automatically during control plane update <<handling_updates>>.
```

Comment:
Suggest:

The {productname} renews all the certificates automatically during control plane update <<handling_updates>>.